### DISCUSSION TITLE
less-loader: can warnings become errors?

### DISCUSSION THREAD

**Discussion by bes** [webpack/webpack#17013] [Q&A]
I would like warnings from `less-loader` to be configurable to become webpack errors.

A typical mistake I see our team make is this, which comes up as a warning:

```
LOG from ./node_modules/less-loader/dist/cjs.js less-loader ./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[6].use[1]!./node_modules/less-loader/dist/cjs.js!./src/main/js/views/visual.module.less
<w> extend ' .body1' has no matches
```

And this happens because someone wrote

```
        div {
            &:extend(.body1);
        }
```

But forgot to import our typography module

```
@import "~lib/system/typography.module.less";
```

Unfortunately, it's very easy to miss this warning, and I would like it to be configurable as an error. I have looked in the `less-loader` documentation but I didn't find anything. Using `WarningsToErrorsPlugin` doesn't work for this case since it's just a log?

I think my real question is, is this a feature request for `less-loader`?

**Comment by alexander-akait** [comment]
It means less outputs this as a warning, I don't think we need a new option, you can create a small plugin:
```js
class WarningsToErrorsPlugin {
  apply(compiler) {
    compiler.hooks.done.tap('WarningsToErrorsPlugin', (stats) => {
      const { warnings } = stats.compilation;
      if (warnings.length > 0) {
        const errorMessage = 'Warnings have been converted to errors.';
        const error = new Error(errorMessage);
        stats.compilation.errors.push(error);
      }
    });
  }
}
```

If you need if you filter them

**Reply by bes** [reply]
Sorry for the _very_ late reply @alexander-akait but here goes -- the problem is that the "Warning" from Less is just a log, as it says in the message:

```
LOG from ./node_modules/less-loader/dist/cjs.js less-loader ./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[6].use[1]!./node_modules/less-loader/dist/cjs.js!......
<w> extend ' .body1' has no matches
<w> extend ' .strong' has no matches
```

It's a LOG, not a real warning. So the plugin you proposed, or the plugin I'm already using `warnings-to-errors-webpack-plugin` does not catch it.

**Reply by bes** [reply]
For example, it would be really nice if the logging facility here (which I guess is what emits the logs) could be configured (optionally) to yield webpack errors if there are warning logs:

https://github.com/webpack-contrib/less-loader/blob/b96340d66e7061102015b01be1c3e44ffa14d30a/src/index.js#L52

Using `this.emitError`

**Reply by alexander-akait** [reply]
Sorry for delay, I see your problem, I will improve less-loader

**Reply by alexander-akait** [reply]
https://github.com/webpack-contrib/less-loader/pull/536

**Reply by bes** [reply]
Amazing 😍

**Reply by alexander-akait** [reply]
Released in `12.1.0`, sorry for delay, a lot of issues and PRs :disappointed:
